### DISCUSSION TITLE
Does deletion of the job rids of stored artefacts and logs?

### DISCUSSION THREAD

**Discussion by hub-dela** [community/community#50934] [Actions] [answered]
### Select Topic Area

Question

### Body

Storing artefacts in workflows using actions/upload-artifact@v3 etc. counts against Package and Storage limit (Paid).
I wonder if deleting workflow run cleans data residue, or said content need to be deliberately cleaned:

example cleanup method:
```
await octokit.rest.actions.deleteWorkflowRun({
          owner: owner,
          repo: repo,
          run_id: run[0]
```

I've confirmed through gh cli and via url, that previously stored artefact is no longer available. 
however, said artefact url refers to workflow, and said negative response not necessary ensures purge.

is it necessary to marshal through workflows and purging artefacts first alternatively, followed by workflow deletion?

additional details:

- in scope of big organisation isolating such clean is complicated, and reporting on store use has delay
- attempt to perform isolated test on private repo failed (not apples to apples), plus it is billed differently from org, plus above mentioned reporting delay

**Comment by enviGit** [comment, marked as answer]
### Hi Denys!
In general when a GitHub Actions job is deleted, the stored artifacts and logs associated with that job will also be deleted automatically. You do not need to manually delete them separately.

However, deleting a workflow run does not immediately remove the artifact from the storage space, as it may take some time for the storage space to be cleared. The duration of time required to remove the storage space depends on the size of the artifact and the usage of the storage space.

If you want to ensure that the artifacts and logs are immediately removed, you can use the `expire_at` parameter in the `actions/upload-artifact@v3` action to specify the duration after which the artifacts should expire. This will automatically remove the artifacts once the specified duration has elapsed.

For more information on managing artifacts in GitHub Actions, you can refer to [the official documentation](https://docs.github.com/en/actions/guides/storing-workflow-data-as-artifacts)

**Reply by hub-dela** [reply]
Yes, this matches my understanding as well and what we normally do, thanks for reply.
However, I am interested in a more intricate detail here, if deleting of the job run immediately result in marking said artefacts for deletion ahead of normal/specified garbage collecting/expiring.

In my particular use case we produce quite a volume of content and forced deletion may prove to be useful. 
Clean on demand vs clean on timer, is what we looking for.

And I am aware of this api
`octokit.rest.actions.deleteArtifact({
  owner,
  repo,
  artifact_id,
});`

though each api call works against token limit (5k calls per hour), hence we looking to reduce number of calls say, with aforementioned workflow run deletion, instead of causing api calls excess by calling both, artefact and run deletion.

**Reply by enviGit** [reply]
Deleting a workflow run will immediately mark the associated artifacts for deletion and they will be removed during the next scheduled cleanup, but the actual deletion may still take some time depending on the size of the artifact and the usage of the storage space. 

Regarding your concern about the API call limit, deleting a workflow run should also delete the associated artifacts, so it may not be necessary to make separate API calls for artifact deletion because as you noted, there is a rate limit for API calls, so you may want to balance your usage of API calls with your need for immediate artifact deletion.

**Reply by hub-dela** [reply]
Hi @enviGit , again, thanks for clarifying the response, I really appreciate your contribution.

While what you say makes total sense, I would really love to see some reference to documentation or api or code, or any other evidence (before/after) confirming this. For now I don't observe immediate effect of the deletion, but that might be the delay.

I am conducting and experiment to verify the statement, but I'd rather have it earlier, before the behaviour will inevitably be checked.

**Reply by hub-dela** [reply]
Thanks for the assist, the behaviour was practically confirmed, and was also discussed and confirmed independently with github representatives
